### PR TITLE
Convert index.html and citing.html to rst.

### PR DESCRIPTION
**Pull request by anntzer**
All's in the title.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are many :doc:`external learning resources <resources/index>` available
+including printed material, videos and tutorials.
+
+Need help?
+~~~~~~~~~~
+
+Matplotlib is a welcoming, inclusive project, and we try to follow the `Python
+Software Foundation Code of Conduct <coc>`_ in everything we do.
```
Comment (dstansby) [inline doc/index.rst]: Should this be `_coc`?

Comment (dstansby) [reply doc/index.rst]: (And similar missing underscores below)

**Comment by tacaswell** [comment]
https://28-7439715-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/index.html looks reasonable.

One thing that having `versions.html` inserted via that script did was that all version of the index page showed the current list of versions (ex http://matplotlib.org/1.3.1/ ).

I have never been sure why we have citing and index as html files.

**Comment by anntzer** [comment]
The link targets were wrong, fixed.

Restored dynamic versions (in a way that works even from the CI renders)... if php works.

File: doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+Documentation
+-------------
+
+This is the documentation for Matplotlib version |version|.
+
+To get started, read the :doc:`User's Guide <users/index>`.
+
+.. raw:: html
+
+   <?php include('http://matplotlib.org/versions.html');?>
```
Comment (jklymak) [inline doc/index.rst]: So, does this work when uploaded?

Comment (jklymak) [reply doc/index.rst]: ping @tacaswell

Comment (jenshnielsen) [reply doc/index.rst]: As far as I know its being served from Github pages and that does not support any server side dynamic pages https://www.quora.com/How-can-I-run-my-PHP-content-on-GitHub-pages

Comment (jklymak) [inline doc/index.rst]: So this needs to change or be fixed

**Comment by anntzer** [comment]
Restored the js snippet.

**Pull request merged**